src/OrchardCore/OrchardCore.Search.AzureAI.Core/Services/AzureAISearchIndexDocumentManager.cs

```diff
@@ -60,13 +60,18 @@ await foreach (var doc in searchResult.Value.GetResultsAsync())
         var searchResult = await client.SearchAsync<SearchDocument>(searchText, searchOptions);
         var counter = 0L;
 
+        if (searchResult.Value is null)
+        {
+            return counter;
+        }
+
         await foreach (var doc in searchResult.Value.GetResultsAsync())
         {
             action(doc.Document);
             counter++;
         }
 
-        return searchResult.Value?.TotalCount ?? counter;
+        return searchResult.Value.TotalCount;
```

Comment:
You still need ` ?? counter;` because the TotalCount will not be set unless its part of the request.

Reply:
I think if the value is set the `TotalCount` will be there, otherwise 0

Reply:
TotalCount is nullable. So it's null when not set.

Reply:
I thought it was not, revert the change